salt/docker/init.sls

```diff
@@ -78,18 +78,22 @@ docker:
     - append_if_not_found: True
     - require:
       - pkg: docker
-  cmd.run:
-    - name: systemctl restart docker.service
-    - onlyif: systemctl status docker.service
-    - onchanges:
-      - /etc/systemd/system/docker.service.d/proxy.conf
-      - /etc/docker/daemon.json
-    - require:
-      - file: /etc/sysconfig/docker
-      - file: /etc/docker/daemon.json
   service.running:
     - enable: True
     - watch:
       - pkg: docker
       - file: /etc/sysconfig/docker
+
+docker-proxy-config:
+  cmd.run:
+    - name: systemctl daemon-reload && systemctl restart docker.service
+    - onlyif: systemctl status docker.service
+    - onchanges:
       - /etc/systemd/system/docker.service.d/proxy.conf
```

Comment:
Should we add `/etc/sysconfig/docker` here too? I guess the docker daemon should be restarted when that file changes...

Reply:
yes, it got lost in all the revisions ;) thanks. only thing is it probably doesnt need a `daemon reload`, so should we put it in yet another section or keep it in the same one?

Reply:
Bah, I would just put it here...

Reply:
If a reload of docker is enough (I don't know), we could just add it to the onchanges section of `docker-reload-config`

Reply:
it needs a restart due to the possible cmd line changes, so a reload is not enough unfortunately.
added it back to the `docker-proxy-config`

Reply:
I agree with @inercia, we can just put it on `docker-proxy-config`... the `daemon-reload` shouldn't be a problem... not needed at all but...